core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -445,14 +445,17 @@ public Block copyRegion(int position, int length)
     @Override
     public boolean mayHaveNull()
     {
-        return positionCount > 0 && dictionary.mayHaveNull();
+        return mayHaveNull;
```

Comment:
Would this result in lazily loaded non-null Dictionary never getting advantage of mayHaveNull ?

Reply:
Check it out now? There is a fast path for mayHaveNull == false and slower one for true.
`&&` is a condition but branch prediction should take care of that